dotnet/src/webdriver/Remote/ResponseLoggerInterceptor.cs

```diff
@@ -0,0 +1,62 @@
+// <copyright file="ResponseLoggerInterceptor.cs" company="WebDriver Committers">
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Net.Http;
+using System.Threading.Tasks;
+using OpenQA.Selenium.Internal.Logging;
+using System;
+
+namespace OpenQA.Selenium.Remote
+{
+    /// <summary>
+    /// Interceptor for logging HTTP responses.
+    /// </summary>
+    public class ResponseLoggerInterceptor : IHttpInterceptor
+    {
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the ResponseLoggerInterceptor with logger.
+        /// </summary>
+        /// <param name="logger">Internal logger to be used for logging responses.</param>
+        public ResponseLoggerInterceptor(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Intercepts an HTTP response and logs the response body if the status code is not in the 2xx.
+        /// </summary>
+        /// <param name="response">The HTTP response message to intercept.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public async Task InterceptAsync(HttpResponseMessage response)
+        {
+            var responseBodyStr = await response.Content.ReadAsStringAsync();
```

Comment:
Should this line not move inside the if statement?

Reply:
should we move this inside?

Reply:
Unless I am missing something it should be inside because we only need the response body inside the if statement. We will be reading the response body every time with the code as it is here even if we don't need it.